relay-server/src/actors/project.rs

```diff
@@ -530,21 +530,33 @@ impl Project {
     fn update_metrics_allowed(&mut self) {
         if let Some(state) = self.state() {
             self.metrics_allowed = state.check_disabled(&self.config).is_ok();
+        } else {
+            // Projects without state go back to the original state of allowing metrics.
+            self.metrics_allowed = true;
```

Comment:
This is only indirectly related, but I think it's more accurate this way.

Reply:
Good catch. Unfortunately, `update_metrics_allowed` doesn't get called yet when the state expires. We'd either have to call this again, or convert the flag into a function that also checks for expiry.

Reply:
Made the flag a function now. Expiry is automatically checked by calling `state()`.